addons/hw_drivers/drivers/SerialBaseDriver.py

```diff
@@ -0,0 +1,183 @@
+        """ Attributes initialization method for `SerialDriver`.
+
+        :param device: path to the device
+        :type device: str
+        """
+
+        super().__init__(device)
+        self._actions = {
+            'get_status': self._get_status,
+        }
+        self._device_connection = 'serial'
+        self._device_lock = Lock()
+        self._status = DeviceStatus()
+        self._status.set_connected()
+        self._set_name()
+
+    @abstractmethod
+    def _get_raw_response(connection):
+        pass
+
+    def _get_status(self):
+        """Updates the current status and pushes it to the frontend."""
+
+        self.data['status'] = self._status.as_dict
+        event_manager.device_changed(self)
+
+    def _set_name(self):
+        """Tries to build the device's name based on its type and protocol name but falls back on a default name if that doesn't work."""
+
+        try:
+            name = ('%s serial %s' % (self._protocol.name, self._device_type)).title()
+        except Exception:
+            name = 'Unknown Serial Device'
+        self._device_name = name
+
+    @abstractmethod
+    def _take_measure(self):
+        pass
+
+    def action(self, data):
+        """Tells the device to perform a specific action.
+
+        :param data: the `_actions` key mapped to the action method we want to call
+        :type data: string
+        """
+
+        def do_action():
+            try:
+                with self._device_lock:
+                    self._actions[data]()
+                    time.sleep(self._protocol.commandDelay)
+            except Exception:
+                msg = _('An error occured while performing action %s on %s') % (data, self.device_name)
+                _logger.exception(msg)
+                self._status.set_error(message_title=msg, message_body=traceback.format_exc())
+                self._get_status()
+
+        if self._connection and self._connection.isOpen():
+            do_action()
+        else:
+            with serial_connection(self.dev.identifier, self._protocol) as connection:
+                self._connection = connection
+                do_action()
+
+    def run(self):
+        """Continuously gets new measures from the device."""
+
+        try:
+            with serial_connection(self.dev.identifier, self._protocol) as connection:
+                self._connection = connection
+                self._status.set_connected()
+                self._get_status()
+                while True:
+                    self._take_measure()
+                    time.sleep(self._protocol.newMeasureDelay)
+        except Exception:
+            msg = _('Error while reading %s') % self.device_name
+            _logger.exception(msg)
+            self._status.set_error(message_title=msg, message_body=traceback.format_exc())
+            self.data['status'] = self._status.as_dict
+            self._get_status()
+
+    @abstractmethod
+    def supported(cls, device=None):
```

Comment:
This method already exists in Driver, and is not abstract. I think it's a quite bad implementation to have a implemented method overridden by an abstract one. You also changed the signature by setting device=None, which change an arg in kwargs, why not handling it in the implementation, or change it in the 'Driver' class